### PR TITLE
Feature/testing pipes

### PR DESCRIPTION
**Pull request by SteffiSchlumpf**
Wrote tests for the pipes.

### CODE REVIEW

File: scenarioo-client/app/app.module.ts [hunk 1]
```diff
@@ -123,7 +123,7 @@ import {BranchAliasesComponent} from './manage/branch-aliases/branch-aliases.com
         DiffInfoService,
         SharePageURL,
         Location,
-        FilterPipe,
+        FilterArrayPipe,
```
Comment (danielsuter) [inline scenarioo-client/app/app.module.ts]: Why do you need to import your pipe as a service?

File: scenarioo-client/app/build/scenarios-overview/scenarios-overview.component.html [hunk 2]
```diff
@@ -38,12 +38,12 @@
 
         <div [ngClass]="isPanelCollapsed ? 'sc-space-top col-sm-12' : 'sc-space-top col-sm-8'">
             <p>
-                <span *ngIf="(scenarios|filter: searchTerm).length === scenarios.length">{{scenarios.length}}</span>
-                <span *ngIf="(scenarios|filter: searchTerm).length != scenarios.length">Found {{(scenarios|filter: searchTerm).length}} of {{scenarios.length}}</span>
+                <span *ngIf="(scenarios|scFilterArray: searchTerm).length === scenarios.length">{{scenarios.length}}</span>
```
Comment (danielsuter) [inline scenarioo-client/app/build/scenarios-overview/scenarios-overview.component.html]: styling is inconsistent - below there are whitespaces before and after the pipe

File: scenarioo-client/app/build/scenarios-overview/scenarios-overview.component.html [hunk 3]
```diff
@@ -38,12 +38,12 @@
...
+                <span *ngIf="(scenarios|scFilterArray: searchTerm).length != scenarios.length">Found {{(scenarios|scFilterArray: searchTerm).length}} of {{scenarios.length}}</span>
```
Comment (danielsuter) [inline scenarioo-client/app/build/scenarios-overview/scenarios-overview.component.html]: !==

File: scenarioo-client/app/build/scenarios-overview/scenarios-overview.component.html [hunk 4]
```diff
@@ -38,12 +38,12 @@
...
                 <span *ngIf="scenarios.length>1"> Scenarios in this Use Case</span>
                 <span *ngIf="scenarios.length<=1"> Scenario in this Use Case</span>
             </p>
-            <table *ngIf="(scenarios|filter: searchTerm).length!=0"
+            <table *ngIf="(scenarios|scFilterArray: searchTerm).length!=0"
```
Comment (danielsuter) [inline scenarioo-client/app/build/scenarios-overview/scenarios-overview.component.html]: !==

File: scenarioo-client/app/build/scenarios-overview/scenarios-overview.component.html [hunk 5]
```diff
@@ -114,7 +114,7 @@
                 </thead>
                 <tbody>
                 <tr (click)="goToScenario(useCaseName, dynamicScenarios.scenario.name)"
-                    *ngFor="let dynamicScenarios of scenarios | orderBy: order:reverse:true | filter: searchTerm
+                    *ngFor="let dynamicScenarios of scenarios | orderBy: order:reverse:true | scFilterArray: searchTerm
```
Comment (danielsuter) [inline scenarioo-client/app/build/scenarios-overview/scenarios-overview.component.html]: why the name dynamicScenario?

File: scenarioo-client/app/build/use-cases-overview/use-cases-overview.component.html [hunk 6]
```diff
@@ -33,12 +33,12 @@
 
         <div [ngClass]="isPanelCollapsed ? 'sc-space-top col-sm-12' : 'sc-space-top col-sm-8'">
             <p>
-                <span *ngIf="(usecases|filter: searchTerm).length === usecases.length">{{usecases.length}}</span>
-                <span *ngIf="(usecases|filter: searchTerm).length !== usecases.length">Found {{(usecases|filter: searchTerm).length}} of {{usecases.length}}</span>
+                <span *ngIf="(usecases|scFilterArray: searchTerm).length === usecases.length">{{usecases.length}}</span>
```
Comment (danielsuter) [inline scenarioo-client/app/build/use-cases-overview/use-cases-overview.component.html]: this looks duplicated with the scenrio view - maybe something to be extracted into a component but I'm not sure if it's really that similar

File: scenarioo-client/app/build/use-cases-overview/use-cases-overview.component.html [hunk 7]
```diff
@@ -107,7 +107,7 @@
                 </thead>
                 <tbody>
                 <tr (click)="goToUseCase(dynamicUseCase.name)"
-                    *ngFor="let dynamicUseCase of usecases | orderBy: order:reverse:true | filter: searchTerm
+                    *ngFor="let dynamicUseCase of usecases | orderBy: order:reverse:true | scFilterArray: searchTerm
```
Comment (danielsuter) [inline scenarioo-client/app/build/use-cases-overview/use-cases-overview.component.html]: why dynamic?

File: scenarioo-client/app/pipes/dateTime.pipe.spec.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+import {DateTimePipe} from './dateTime.pipe';
+
+declare var angular: angular.IAngularStatic;
+
+describe('Pipe: scDateTime', () => {
+    let scDateTime: DateTimePipe;
+    let input;
+
+    // Arrange: set up new instance of the pipe DateTime
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/dateTime.pipe.spec.ts]: remove comment

File: scenarioo-client/app/pipes/dateTime.pipe.spec.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {DateTimePipe} from './dateTime.pipe';
+
+declare var angular: angular.IAngularStatic;
+
+describe('Pipe: scDateTime', () => {
+    let scDateTime: DateTimePipe;
+    let input;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/dateTime.pipe.spec.ts]: string?

File: scenarioo-client/app/pipes/dateTime.pipe.spec.ts [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+    // Arrange: set up new instance of the pipe DateTime
+    beforeEach(() => {
+        scDateTime = new DateTimePipe('en-US');
+    });
+
+    it('Undefined returns an empty string', async () => {
+        // Act
+        input = scDateTime.transform(undefined);
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/dateTime.pipe.spec.ts]: input is actually output or I don't get it. You put input into your pipe and get output which you then assert

File: scenarioo-client/app/pipes/dateTime.pipe.ts [hunk 11]
```diff
@@ -1,6 +1,8 @@
 import {Injectable, Pipe, PipeTransform} from '@angular/core';
 import {DatePipe} from '@angular/common';
 
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/dateTime.pipe.ts]: please remove that - old AngularJS code must not be part of new components and I don't see any reference to it in the code

File: scenarioo-client/app/pipes/filterArray.pipe.spec.ts [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {FilterArrayPipe} from './filterArray.pipe';
+
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/filterArray.pipe.spec.ts]: remove

File: scenarioo-client/app/pipes/dateTime.pipe.spec.ts [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {DateTimePipe} from './dateTime.pipe';
+
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/dateTime.pipe.spec.ts]: remove

File: scenarioo-client/app/pipes/filterArray.pipe.spec.ts [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+            something: 'else',
+            odd: {
+                weird: 'things',
+            },
+        },
+    ];
+
+    const MODEL_EMPTY = [];
+
+    // Arrange: set up new instance of the pipe FilterArray
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/filterArray.pipe.spec.ts]: remove comment

File: scenarioo-client/app/pipes/filterArray.pipe.spec.ts [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+    beforeEach(() => {
+        scFilterArray = new FilterArrayPipe();
+    });
+
+    it('Should return the original model when search text is empty', async () => {
+        // Act
+        input = scFilterArray.transform(MODEL, '');
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/filterArray.pipe.spec.ts]: naming like above - call it output or actual

File: scenarioo-client/app/pipes/filterArray.pipe.ts [hunk 16]
```diff
@@ -1,11 +1,13 @@
 import { Injectable, Pipe, PipeTransform } from '@angular/core';
 
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/filterArray.pipe.ts]: remove

File: scenarioo-client/app/pipes/humanReadable.pipe.spec.ts [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {HumanReadablePipe} from './humanReadable.pipe';
+
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/humanReadable.pipe.spec.ts]: remove

File: scenarioo-client/app/pipes/humanReadable.pipe.spec.ts [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+
+describe('Pipe: scHumanReadable', () => {
+    let scHumanReadable: HumanReadablePipe;
+    let input;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/humanReadable.pipe.spec.ts]: naming

File: scenarioo-client/app/pipes/humanReadable.pipe.ts [hunk 19]
```diff
@@ -1,5 +1,7 @@
 import {Injectable, Pipe, PipeTransform} from '@angular/core';
 
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/humanReadable.pipe.ts]: remove

File: scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.spec.ts [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {MetadataTreeCreatorPipe} from './metadataTreeCreator.pipe';
+import {TreeDataOptimizerPipe} from './treeDataOptimizer.pipe';
+import {TreeDataCreatorPipe} from './treeDataCreator.pipe';
+
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.spec.ts]: remove

File: scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.spec.ts [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+
+describe('Pipe: scMetadataTreeCreator', () => {
+    let scMetadataTreeCreator: MetadataTreeCreatorPipe;
+    let input;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.spec.ts]: naming

File: scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.spec.ts [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+            },
+        ],
+    };
+
+    // Arrange: set up new instance of the pipe MetadataTreeCreator
+    beforeEach(() => {
+        scMetadataTreeCreator = new MetadataTreeCreatorPipe(new TreeDataOptimizerPipe(), new TreeDataCreatorPipe());
+    });
+
+    it('transforms javascript object into an optimized tree', async () => {
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.spec.ts]: think of some edge cases tests (e.g. undefined values, mixture of strings etc.)

File: scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.ts [hunk 23]
```diff
@@ -2,6 +2,8 @@ import {Injectable, Pipe, PipeTransform} from '@angular/core';
 import {TreeDataOptimizerPipe} from './treeDataOptimizer.pipe';
 import {TreeDataCreatorPipe} from './treeDataCreator.pipe';
 
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.ts]: remove

File: scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.spec.ts [hunk 24]
```diff
@@ -0,0 +1,62 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {MetadataTreeListCreatorPipe} from './metadataTreeListCreator.pipe';
+import {MetadataTreeCreatorPipe} from './metadataTreeCreator.pipe';
+import {TreeDataOptimizerPipe} from './treeDataOptimizer.pipe';
+import {TreeDataCreatorPipe} from './treeDataCreator.pipe';
+
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.spec.ts]: remove

File: scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.spec.ts [hunk 25]
```diff
@@ -0,0 +1,62 @@
...
+            childNodes: [
+                {
+                    nodeLabel: 'theAnswer',
+                    nodeValue: 42,
+                },
+            ],
+        },
+    };
+
+    // Arrange: set up new instance of the pipe MetadataTreeListCreator
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.spec.ts]: remove comment

File: scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.ts [hunk 26]
```diff
@@ -1,6 +1,7 @@
 import {Injectable, Pipe, PipeTransform} from '@angular/core';
 import {MetadataTreeCreatorPipe} from './metadataTreeCreator.pipe';
-import {IDetailable, IUseCase} from '../generated-types/backend-types';
+
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.ts]: remove

File: scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.ts [hunk 27]
```diff
@@ -12,21 +13,13 @@ export class MetadataTreeListCreatorPipe implements PipeTransform {
     constructor(private metadataTreeCreatorPipe: MetadataTreeCreatorPipe) {
     }
 
-    transform(metadata: any): any[] {
+    transform(metadata: any): any {
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.ts]: it returns an array of object if I'm not wrong

File: scenarioo-client/app/pipes/metadata/treeDataCreator.pipe.spec.ts [hunk 28]
```diff
@@ -0,0 +1,190 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {TreeDataCreatorPipe} from './treeDataCreator.pipe';
+
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/treeDataCreator.pipe.spec.ts]: remove

File: scenarioo-client/app/pipes/metadata/treeDataCreator.pipe.spec.ts [hunk 29]
```diff
@@ -0,0 +1,190 @@
...
+
+describe('Pipe: scTreeDataCreator', () => {
+    let scTreeDataCreator: TreeDataCreatorPipe;
+    let input;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/treeDataCreator.pipe.spec.ts]: naming

File: scenarioo-client/app/pipes/metadata/treeDataCreator.pipe.spec.ts [hunk 30]
```diff
@@ -0,0 +1,190 @@
...
+
+    const CHILDNODES_WITHOUT_NODELABEL_TRANSFORMED = {
+        nodeLabel: '',
+        childNodes: [
+            {nodeLabel: 'Listentry 0'},
+            {nodeLabel: 'Listentry 1'},
+        ],
+    };
+
+    // Arrange: set up new instance of the pipe TreeDataCreator
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/treeDataCreator.pipe.spec.ts]: remove comment

File: scenarioo-client/app/pipes/metadata/treeDataCreator.pipe.ts [hunk 31]
```diff
@@ -1,5 +1,7 @@
 import {Injectable, Pipe, PipeTransform} from '@angular/core';
 
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/treeDataCreator.pipe.ts]: remove

File: scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts [hunk 32]
```diff
@@ -0,0 +1,323 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {TreeDataOptimizerPipe} from './treeDataOptimizer.pipe';
+
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts]: remove

File: scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts [hunk 33]
```diff
@@ -0,0 +1,323 @@
...
+
+describe('Pipe: scTreeDataOptimizer', () => {
+    let scTreeDataOptimizer: TreeDataOptimizerPipe;
+    let input;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts]: naming

File: scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts [hunk 34]
```diff
@@ -0,0 +1,323 @@
...
+        nodeLabel: 'Item',
+        nodeValue: 'someValue',
+    };
+
+    // Arrange: set up new instance of the pipe TreeDataOptimizer
+    beforeEach(() => {
+        scTreeDataOptimizer = new TreeDataOptimizerPipe();
+    });
+
+    // TODO Check with Rolf whether we need to remove empty child nodes
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts]: please check with @bruderol  :-)

File: scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.ts [hunk 35]
```diff
@@ -1,5 +1,7 @@
 import {Injectable, Pipe, PipeTransform} from '@angular/core';
 
+declare var angular: angular.IAngularStatic;
```
Comment (danielsuter) [inline scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.ts]: remove

File: scenarioo-client/app/app.module.ts (hunk 1 above)
Comment (SteffiSchlumpf) [reply scenarioo-client/app/app.module.ts]: I need to inject the pipe into a class, therefore I provide it in the providers array of ngModule. I read that in the angular documentation (https://angular.io/guide/pipes#pipes). Do you see a better way?

File: scenarioo-client/app/build/use-cases-overview/use-cases-overview.component.html (hunk 6 above)
Comment (SteffiSchlumpf) [reply scenarioo-client/app/build/use-cases-overview/use-cases-overview.component.html]: You're right. I will start a new branch to divide the table into several components as soon as I'm finished with the detailarea...

File: scenarioo-client/app/pipes/dateTime.pipe.spec.ts (hunk 10 above)
Comment (SteffiSchlumpf) [reply scenarioo-client/app/pipes/dateTime.pipe.spec.ts]: You're right, thanks.

File: scenarioo-client/app/build/use-cases-overview/use-cases-overview.component.html (hunk 7 above)
Comment (SteffiSchlumpf) [reply scenarioo-client/app/build/use-cases-overview/use-cases-overview.component.html]: The usecases are dynamically loaded into the table? Which name do you suggest?

File: scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.ts (hunk 27 above)
Comment (SteffiSchlumpf) [reply scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.ts]: `transform(metadata: any): Array<object>`  is not working

File: scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.spec.ts (hunk 22 above)
Comment (SteffiSchlumpf) [reply scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.spec.ts]: Isn't that already tested in treeDataCreator.pipe.spec.ts?

File: scenarioo-client/app/build/use-cases-overview/use-cases-overview.component.html (hunk 7 above)
Comment (bruderol) [reply scenarioo-client/app/build/use-cases-overview/use-cases-overview.component.html]: I would suggest just to use `usecase` - it is obvious from context that this gets somehow loaded or injected dynamically, this does not have to be in the name otherwise we would have to name every thing `dynamicXy` that is dynamically loaded.

File: scenarioo-client/app/pipes/filterArray.pipe.ts [hunk 36]
```diff
@@ -1,11 +1,11 @@
 import { Injectable, Pipe, PipeTransform } from '@angular/core';
 
 @Pipe({
-    name: 'filter',
+    name: 'scFilterArray',
 })
 
 @Injectable()
-export class FilterPipe implements PipeTransform {
+export class FilterArrayPipe implements PipeTransform {
```
Comment (bruderol) [inline scenarioo-client/app/pipes/filterArray.pipe.ts]: I think it would be great to have a short documentation as a comment on each important component or pipe about what the intention of this pipe is - what is its responsibility (see also RDD Responsibility Driven Design) - Also I think the name is not very intuitive - it is to filter by a search term, right? So maybe something like "SearchFilterPipe" would be a better name?

File: scenarioo-client/app/pipes/humanReadable.pipe.spec.ts [hunk 37]
```diff
@@ -0,0 +1,72 @@
...
+        await expect(output).toContain('&/%');
+        await expect(output).toContain('String');
+    });
+
+    it('Should replace underline with blanks', async () => {
+        // Act
+        output = scHumanReadable.transform('This_may_also_be_acceptable');
+        // Assert
+        await expect(output).toEqual('This may also be acceptable');
+    });
```
Comment (bruderol) [inline scenarioo-client/app/pipes/humanReadable.pipe.spec.ts]: Very cool test. Very well readable 👍 
well done!

File: scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.spec.ts (hunk 22 above)
Comment (bruderol) [reply scenarioo-client/app/pipes/metadata/metadataTreeCreator.pipe.spec.ts]: Can you put just one other edge case into the example - with two keys - call one "Key Can Be Human Readable in the Data as Written By Tests" and "keyIsNotTransformedToHumanReadableAndDisplayedAsIs" - which was one case we discussed a long time about it until we realized that it has to be like that. Maybe the test helps to understand this in future better. You could also put a comment that it was decided not to make these strings human readable since clients can write it human readbale when they want to have it like that. Thanks.

File: scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.ts (hunk 27 above)
Comment (bruderol) [reply scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.ts]: try `any[]` instead

Comment (bruderol) [reply scenarioo-client/app/pipes/metadata/metadataTreeListCreator.pipe.ts]: oh, I see, it is not returning an array, but really a generic object tree, then `any` is correct. Sorry.

File: scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts (hunk 34 above)
Comment (bruderol) [reply scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts]: I think this case is irrelevant - if a user writes weird stuff into its data tress, its going to be displayed weired - we do not have to care about this case - so let's remove it.

File: scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts [hunk 38]
```diff
@@ -0,0 +1,320 @@
...
+                nodeLabel: 'details',
+                childNodes: [
+                    {
+                        nodeLabel: 'hero',
+                        nodeValue: 'Donald Duck',
+                    },
+                ],
+            },
+        ],
+    };
```
Comment (bruderol) [inline scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts]: Just a proposal to improve readability of the tests (I know this is not your fault -as you did not write this code in a first place): Since all of this constants are just used in one test I would rather prefer to define the consts directly in the place where they are used (thus inside the tests), as it keeps the tests more readable directly in one place without having to jump forth an back.

File: scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts (hunk 34 above)
Comment (bruderol) [reply scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts]: As resolved together with you

File: scenarioo-client/app/pipes/filterArray.pipe.ts (hunk 36 above)
Comment (bruderol) [reply scenarioo-client/app/pipes/filterArray.pipe.ts]: done together
